src/Symfony/Component/VarDumper/Caster/SocketCaster.php

```diff
@@ -0,0 +1,42 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\VarDumper\Caster;
+
+use Symfony\Component\VarDumper\Cloner\Stub;
+
+/**
+ * @author Nicolas Grekas <p@tchwork.com>
+ */
+final class SocketCaster
+{
+    public static function castSocket(\Socket $h, array $a, Stub $stub, bool $isNested): array
+    {
+        if (socket_atmark($h)) {
```

Comment:
`socket_atmark` does not exist on PHP 8.2, while Socket exists there. We should have a `\PHP_VERSION_ID >= 80300` check.

Reply:
ah right, updated